lib/themes/utils/fontSizeRulesForTokens.js

```diff
@@ -2,66 +2,71 @@ import merge from 'lodash/merge';
 import alignTextToGrid from './alignTextToGrid';
 import createResponsiveRules from './createResponsiveRules';
 import { px } from './toUnit';
+import each from 'lodash/each';
 
 export default ({ tokens }) => {
   const rules = [];
-  Object.keys(tokens.text).forEach(typeName => {
-    const type = tokens.text[typeName];
+  each(
+    {
+      ...tokens.text,
+      ...tokens.heading
+    },
+    (type, typeName) => {
+      const {
+        fontSize: mobileFontSize,
+        lineHeight: mobileLineHeight
+      } = alignTextToGrid(type.mobile, tokens);
+      const {
+        fontSize: desktopFontSize,
+        lineHeight: desktopLineHeight
+      } = alignTextToGrid(type.desktop, tokens);
 
-    const {
-      fontSize: mobileFontSize,
-      lineHeight: mobileLineHeight
-    } = alignTextToGrid(type.mobile, tokens);
-    const {
-      fontSize: desktopFontSize,
-      lineHeight: desktopLineHeight
-    } = alignTextToGrid(type.desktop, tokens);
-
-    const mobileRules = {
-      fontSize: mobileFontSize,
-      lineHeight: mobileLineHeight
-    };
-    const desktopRules = {
-      fontSize: desktopFontSize,
-      lineHeight: desktopLineHeight
-    };
-
-    rules.push(
-      createResponsiveRules({
-        tokens,
-        selector: `.${typeName}`,
-        mobileRules,
-        desktopRules
-      })
-    );
-
-    if (typeName === 'standard') {
-      const interactionHeight = tokens.interactionRows * tokens.rowHeight;
-      const mobileInteractionPadding = px(
-        (interactionHeight - type.mobile.rows * tokens.rowHeight) / 2
-      );
-      const desktopInteractionPadding = px(
-        (interactionHeight - type.desktop.rows * tokens.rowHeight) / 2
-      );
+      const mobileRules = {
+        fontSize: mobileFontSize,
+        lineHeight: mobileLineHeight
+      };
+      const desktopRules = {
+        fontSize: desktopFontSize,
+        lineHeight: desktopLineHeight
+      };
 
       rules.push(
         createResponsiveRules({
           tokens,
-          selector: '.interaction',
-          mobileRules: {
-            ...mobileRules,
-            paddingTop: mobileInteractionPadding,
-            paddingBottom: mobileInteractionPadding
-          },
-          desktopRules: {
-            ...desktopRules,
-            paddingTop: desktopInteractionPadding,
-            paddingBottom: desktopInteractionPadding
-          }
+          selector: `.${typeName}`,
+          mobileRules,
+          desktopRules
         })
       );
+
+      if (typeName === 'standard') {
+        const interactionHeight = tokens.interactionRows * tokens.rowHeight;
+        const mobileInteractionPadding = px(
+          (interactionHeight - type.mobile.rows * tokens.rowHeight) / 2
```

Comment:
Dividing always makes me 😰 - how does this behave if it's an odd value on the left hand side?

Reply:
I hear you, in this case partial pixel values should not matter as by applying the difference between `interactionHeight` and the computer type height equally above and below the text, it will have the desired outcome of vertically centering within the touch target.